core/modules/cart/store/mutations.ts

```diff
@@ -56,7 +57,7 @@ const mutations: MutationTree<CartState> = {
     let record = state.cartItems.find(p => (productsEquals(p, product) || (p.server_item_id && p.server_item_id === product.server_item_id)))
     if (record) {
       EventBus.$emit('cart-before-itemchanged', { item: record })
-      record = Object.assign(record, product)
+      Object.entries(product).forEach(([key, value]) => Vue.set(record, key, value))
```

Comment:
it shouldn't be such a logic inside of mutation...  but I understand that is the only solution